lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -250,21 +256,23 @@ export class PublicClientApplication {
      * any code that follows this function will not execute.
      * @param {@link (AuthenticationParameters:type)}
      */
-    loginRedirect(request: AuthenticationParameters): void {
+    async loginRedirect(request: AuthorizationUrlRequest): Promise<void> {
         // Preflight request
-        this.preflightRequest();
-
+        const validRequest: AuthorizationUrlRequest = this.preflightRequest(request);
         try {
+            // Create auth code request and generate PKCE params
+            const authCodeRequest: AuthorizationCodeRequest = await this.generateAuthorizationCodeRequest(validRequest);
```

Comment:
This is good for now. When we add `interactiveFlow` which would need `PKCE generation` for `msal-node` we can think of migrating this to `msal-common` or have equivalent code in `msal-node`.

Reply:
cc @sangonzal we need to track this for `msal-node`